File: elementary_computer_science/Python/square.py
inf = 1000
xmin, xmax, ymin, ymax = inf, -inf, inf, -inf
file = open("square.inp", "r")
n = int(file.readline())
for _ in range(n):
	x, y = map(int, file.readline().split())
	xmin = min(xmin, x)
	xmax = max(xmax, x)
	ymin = min(ymin, y)
	ymax = max(ymax, y)
file.close()
file = open("square.out", "w")
c = max(xmax - xmin, ymax - ymin)
file.write(str(c*c))
file.close()

File: elementary_computer_science/Cpp/square.cpp
#include <bits/stdc++.h>
using namespace std;
#define inf 1000
int n, xmin = inf, ymin = inf, xmax = -inf, ymax = -inf;
int main() {
	freopen("square.inp", "r", stdin);
	freopen("square.out", "w", stdout);
	cin >> n;
	for (int i = 1; i <= n; ++i) {
		int x, y;
		cin >> x >> y;
		xmin = min(xmin, x);
		xmax = max(xmax, x);
		ymin = min(ymin, y);
		ymax = max(ymax, y);
	}
	int c = max(xmax - xmin, ymax - ymin);
	cout << c*c;
}